src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -311,10 +314,12 @@ public void close() {
     K key = _keyDeserializer.deserialize(tp.topic(), consumerRecord.key());
     byte[] valueBytes = parseAndMaybeTrackRecord(tp, consumerRecord.offset(), consumerRecord.value());
     V value = _valueDeserializer.deserialize(tp.topic(), valueBytes);
-    if (value != null) {
+    if (valueBytes != INCOMPLETE_RESULT) {
```

Comment:
The handling of the value are user code. We should use whatever value returned by the deserializer. It would be a little weird to have any value other than that. Also we do not call any method on the value so it is safe.